app/assets/javascripts/dashboard/current_tasks.js

```diff
@@ -1,16 +1,77 @@
-/* global dropdownSelector */
+/* global dropdownSelector I18n animateSpinner PerfectSb */
 /* eslint-disable no-param-reassign */
 
 var DasboardCurrentTasksWidget = (function() {
-  function initFilters() {
-    var sortFilter = '.curent-tasks-filters .sort-filter';
-    var viewFilter = '.curent-tasks-filters .view-filter';
-    var projectFilter = '.curent-tasks-filters .project-filter';
-    var experimentFilter = '.curent-tasks-filters .experiment-filter';
+  var sortFilter = '.curent-tasks-filters .sort-filter';
+  var viewFilter = '.curent-tasks-filters .view-filter';
+  var projectFilter = '.curent-tasks-filters .project-filter';
+  var experimentFilter = '.curent-tasks-filters .experiment-filter';
+  var emptyState = `<div class="no-tasks">
+                      <p class="text-1">${ I18n.t('dashboard.current_tasks.no_tasks.text_1') }</p>
+                      <p class="text-2">${ I18n.t('dashboard.current_tasks.no_tasks.text_2') }</p>
+                      <i class="fas fa-angle-double-down"></i>
+                    </div>`;
 
-    $('.curent-tasks-filters .clear-button').click(() => {
+  function loadCurrentTasksList() {
+    var $currentTasksList = $('.current-tasks-list');
+    var params = {
+      project_id: dropdownSelector.getValues(projectFilter),
+      experiment_id: dropdownSelector.getValues(experimentFilter),
+      sort: dropdownSelector.getValues(sortFilter),
+      view: dropdownSelector.getValues(viewFilter),
+      mode: $('.current-tasks-navbar .active').data('mode')
+    };
+    animateSpinner($currentTasksList, true);
+    $.get($currentTasksList.attr('data-tasks-list-url'), params, function(data) {
+      // Toggle empty state
+      if (data.tasks_list.length === 0) {
+        $currentTasksList.append(emptyState);
+      } else {
+        $currentTasksList.find('.no-tasks').remove();
+      }
+      // Clear the list
+      $currentTasksList.find('.current-task-item').remove();
```

Comment:
I don't think you can combine it, because sometimes you need, when you switch tabs(all/assigned) to display just empty-placeholder and remove all tasks. It means you need different classes to toggle between these two options.